lib/src/op_heads_store.rs

```diff
@@ -121,20 +121,25 @@ pub fn resolve_op_heads<E>(
     )?;
     let op_head_ids_after: HashSet<_> =
         filtered_op_heads.iter().map(|op| op.id().clone()).collect();
-    for removed_op_head in op_head_ids_before.difference(&op_head_ids_after) {
-        op_heads_store.remove_op_head(removed_op_head);
-    }
+    let ancestor_op_heads = op_head_ids_before
+        .difference(&op_head_ids_after)
+        .cloned()
+        .collect_vec();
     let mut op_heads = filtered_op_heads.into_iter().collect_vec();
 
     // Return without creating a merge operation
     if op_heads.len() == 1 {
-        return Ok(op_heads.pop().unwrap());
+        let op_head = op_heads.pop().unwrap();
```

Comment:
nit: could be `if let [op_head] = &*op_heads`, but it wouldn't work if `op_heads` were `HashSet`. I don't know which is better.

Reply:
Done.